repo/blob/storage.go

```diff
@@ -19,6 +19,10 @@ var ErrSetTimeUnsupported = errors.Errorf("SetTime is not supported")
 // ErrInvalidRange is returned when the requested blob offset or length is invalid.
 var ErrInvalidRange = errors.Errorf("invalid blob offset or length")
 
+// ErrTokenExpired is returned when the token used for
+// authenticating with storage provider has expired.
+var ErrTokenExpired = errors.Errorf("The provided token has expired")
```

Comment:
Is the intention to return this error from a specific (blob) storage implementation?

Reply:
@julio-lopez  - This is the string returned in the error from AWS when blob related operations are performed using an expired token. I assumed that the check I added in isRetriable() will compare the `err` returned from the provider with the blob.errTokenExpired I defined in storage.go .  Would that check in isRetriable work ? (I havent had a chance to build the images and test this yet. I will try to get to in the current or next sprints)

Reply:
IIUC, `errors.Is(...)` won't return `true` if the error messages match. The actual comparison is based on whether 
- the types match or can be assigned to the provided error; or
- the target error implements an `Is(err error) bool` function that would return true for the specific case.
https://pkg.go.dev/errors#Is

This particular case needs an additional "translation" from the (S3) store specific error to the "ErrTokenExpired" error when the error message matches. See examples in the s3 package.

Can you add unit tests to check the error matching functionality?

Reply:
Thank you for the pointer @julio-lopez  . I've updated the PR. I'll post test results in the PR's description.